### DISCUSSION TITLE
Question regarding one of the examples

### DISCUSSION THREAD

**Discussion by clarknova** [floating-ui/floating-ui#2695] [General]
I am looking at this example: [Link](https://codesandbox.io/p/sandbox/admiring-lamport-5wt3yg?file=%2Fsrc%2FDropdownMenu.tsx%3A122%2C4-126%2C6)

I am trying to figure out what the purpose of this code is
```
 function onSubMenuOpen(event: { nodeId: string; parentId: string }) {
      if (event.nodeId !== nodeId && event.parentId === parentId) {
        setIsOpen(false);
      }
```

I can remove it and it does not seem the affect anything.
As far as i understand the useHover interaction already closes submenus?

**Comment by atomiks** [comment]
It prevents multiple submenus from being open simultaneously with a mix of keyboard and mouse interaction.

While hovering over the last submenu, use the arrow keys to open the submenu just above it. Then move the mouse a bit so it opens the last submenu.
